File: ios/videoplayer/VideoPlayerManager.m
//
//  VideoPlayerModule.m
//  videoplayer
//
//  Created by Cong Do Nguyen on 4/20/23.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(VideoPlayerManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(videoUrl, NSString)
RCT_EXPORT_VIEW_PROPERTY(onLoaded, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCurrentTimeChange, RCTDirectEventBlock)
RCT_EXTERN_METHOD(pause)
RCT_EXTERN_METHOD(play)
RCT_EXTERN_METHOD(seek: (float *) to)
@end

File: ios/videoplayer/VideoPlayerManager.swift
//
//  VideoPlayerModule.swift
//  videoplayer
//
//  Created by Cong Do Nguyen on 4/20/23.
//

import Foundation
import AVFoundation

@objc(VideoPlayerManager)
class VideoPlayerManager: RCTViewManager {
  var videoPlayerView: VideoPlayerView?
  
  override func view() -> UIView! {
    let myView = VideoPlayerView()
    videoPlayerView = myView
    return myView
  }
  
  @objc func pause() {
    videoPlayerView?.pause()
  }
  
  @objc func play() {
    videoPlayerView?.continueVideo()
  }
  
  @objc func seek(_ to: Float) {
    videoPlayerView?.seek(to: Double(to))
  }
  
  @objc
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
